### PR TITLE
Swaps: Use quotes conversion rate

### PR DESCRIPTION
**Pull request by wachunei**
**Description**

From https://github.com/MetaMask/swaps-controller/issues/41

> To calculate the value of the quotes the controller needs the destination token conversion rate, this is passed from the client to the controller, forcing the client to have it added to their wallet everytime.
> 
> Upcoming version of the API includes both the source and destination rate values as part of the response:
> ```ts
> export interface Quote {
> // ...
> sourceTokenRate: number | null;
> destinationTokenRate: number | null;
> // ...
> }
> ```
> 
> These should be used instead of the values coming from the client.

This PR **removes having to add the destination token to fetch quotes** and only adds it if a swaps is attempted.

**TODO**
- [x] Remove local file with swaps-controller from https://github.com/MetaMask/swaps-controller/pull/43 once published

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: app/components/UI/Swaps/QuotesView.js [hunk 1]
```diff
@@ -601,6 +580,23 @@ function SwapsQuotesView({
...
+			}
+			if (
+				!isSwapsNativeAsset(sourceToken) &&
+				!AssetsController.state.tokens.includes(token => toLowerCaseCompare(token.address, sourceToken.address))
+			) {
+				const { address, symbol, decimals } = sourceToken;
+				await AssetsController.addToken(address, symbol, decimals);
+			}
 		} catch (e) {
 			// send analytics
```
Comment (sethkfman) [inline app/components/UI/Swaps/QuotesView.js:601]: Should we log this failure?

File: app/components/UI/Swaps/QuotesView.js [hunk 2]
```diff
@@ -601,6 +580,23 @@ function SwapsQuotesView({
 				WalletDevice.MM_MOBILE
 			);
 			updateSwapsTransactions(transactionMeta, approvalTransactionMetaId, newSwapsTransactions);
+
+			if (
```
Comment (sethkfman) [inline app/components/UI/Swaps/QuotesView.js:584]: Could you make this a common function and just pass in the token to reduce common lines?

**Review by sethkfman — COMMENTED** [review]
Just a couple of minor comments

File: app/components/UI/Swaps/QuotesView.js (hunk 1 above)
Comment (wachunei) [reply app/components/UI/Swaps/QuotesView.js:601]: I'm not sure why this is done this way, I'm not familiar with any possible exception we can run into here. @estebanmino do you have more info?

File: app/components/UI/Swaps/QuotesView.js (hunk 2 above)
Comment (wachunei) [reply app/components/UI/Swaps/QuotesView.js]: done!

**Review by sethkfman — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍🏽

**Pull request merged**